From 1587828e06491339654d20dea3382749fde143ed Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Mon, 29 Jan 2024 11:02:06 +0800
Subject: [PATCH 11/21] wifi: mt76: fix tx statistics about tx retry and tx
 fail

The tx retry and tx failed are reported by PPDU TxS.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 mt76_connac_mac.c | 3 ---
 mt7915/mac.c      | 2 +-
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/mt76_connac_mac.c b/mt76_connac_mac.c
index b841bf62..630c6402 100644
--- a/mt76_connac_mac.c
+++ b/mt76_connac_mac.c
@@ -716,9 +716,6 @@ bool mt76_connac2_mac_add_txs_skb(struct mt76_dev *dev, struct mt76_wcid *wcid,
 	struct sk_buff_head list;
 	struct sk_buff *skb;
 
-	if (le32_get_bits(txs_data[0], MT_TXS0_TXS_FORMAT) == MT_TXS_PPDU_FMT)
-		return false;
-
 	mt76_tx_status_lock(dev, &list);
 	skb = mt76_tx_status_skb_get(dev, wcid, pid, &list);
 	if (skb) {
diff --git a/mt7915/mac.c b/mt7915/mac.c
index e167e7b6..a5d0b096 100644
--- a/mt7915/mac.c
+++ b/mt7915/mac.c
@@ -1021,7 +1021,7 @@ static void mt7915_mac_add_txs(struct mt7915_dev *dev, void *data)
 
 	msta = container_of(wcid, struct mt7915_sta, wcid);
 
-	if (pid == MT_PACKET_ID_WED)
+	if (le32_get_bits(txs_data[0], MT_TXS0_TXS_FORMAT) == MT_TXS_PPDU_FMT)
 		mt76_connac2_mac_fill_txs(&dev->mt76, wcid, txs_data);
 	else
 		mt76_connac2_mac_add_txs_skb(&dev->mt76, wcid, pid, txs_data);
-- 
2.18.0

